app/controllers/carriers_controller.rb

```diff
@@ -0,0 +1,41 @@
+class CarriersController < ApplicationController
+  after_action :log
+
+  def calculate
+    zip = params[:zip]
+    city = params[:city]
+    state = params[:state]
+    items = params[:items]
+    begin
+        usps = Carrier.estimate_usps_shipping(items, state, city, zip)
+        ups = Carrier.estimate_ups_shipping(items, state, city, zip)
+        estimate = {
+          "usps" => usps,
+          "ups" => ups
+        }
+        render json: estimate.as_json, :status => :ok
+    rescue ActiveShipping::ResponseError
+        render json: [], :status => :bad_request
+    end
```

Comment:
Thank you! Charles helped us out with this one, so it's still kinda confusing to me.